fenix/app/src/main/java/org/mozilla/fenix/components/appstate/AppStoreReducer.kt

```diff
@@ -234,6 +234,18 @@ internal object AppStoreReducer {
         )
 
         is AppAction.ShoppingAction -> ShoppingStateReducer.reduce(state, action)
+
+        is AppAction.AwesomeBarAction.VisibilityStateUpdated -> {
+            val awesomeBarState = state.awesomeBarState.copy(visibilityState = action.visibilityState)
+            state.copy(awesomeBarState = awesomeBarState)
+        }
+        is AppAction.AwesomeBarAction.SuggestionClicked -> {
+            val awesomeBarState = state.awesomeBarState.copy(clickedSuggestion = action.suggestion)
+            state.copy(awesomeBarState = awesomeBarState)
+        }
+        is AppAction.AwesomeBarAction.EngagementFinished -> {
+            state.copy(awesomeBarState = AppState.AwesomeBarState())
```

Comment:
nit: This is a build failure for me locally.

```suggestion
            state.copy(awesomeBarState = AwesomeBarState())
```